### PR TITLE
[spirv] support SPV_EXT_shader_image_int64 extension for 64-bit integer resource types

### PR DESCRIPTION
**Pull request by LukasBanana**
This commit adds `Int64ImageEXT` capability and `SPV_EXT_shader_image_int64` extension to the SPIRV backend when 64-bit integer types are used on an image resource.

### CODE REVIEW

**Comment by jaebaek** [comment]
Thank you for your contribution!
Mostly looks good to me, but could you please add a unit test similar to [diff.txt](https://github.com/microsoft/DirectXShaderCompiler/files/6536041/diff.txt)?

**Comment by jaebaek** [comment]
Both are fine but we want to put it into this PR. When you submit your commit, you have to use "Squash and merge" (the submission button). It will be finally just a single commit :)

By the way the `SPV_EXT_shader_image_int64` spec says it needs SPIR-V 1.3, but the corresponding Vulkan extension doesn't require Vulkan 1.1. Since glslang implementation https://github.com/KhronosGroup/glslang/pull/2409 does not require SPIR-V 1.3, I guess it is just a spec bug and we can use it without considering the SPIR-V version.

Anyway, I will approve this PR after you add the unit test.

**Comment by LukasBanana** [comment]
@jaebaek I added the unit test to the previous commit and force pushed to our fork as that seemed to be the easiest way for me.

**Review by jaebaek — APPROVED** [review]

**Comment by jaebaek** [comment]
Thanks! Could you please sign this CLA https://cla.opensource.microsoft.com/microsoft/DirectXShaderCompiler?pullRequest=3787 ?

**Comment by LukasBanana** [comment]
@jaebaek sure thing, I just have to wait for a response from our legal team as I have to sign it in the name of my employer. Hopefully not later than earlier next week.

**Comment by jaebaek** [comment]
Thanks :)

**Comment by LukasBanana** [comment]
@jaebaek I signed the CLA :)

**Comment by jaebaek** [comment]
Nice. Thank you again for your contribution!

**Pull request merged**
